tests/test_stepreporter.py

```diff
@@ -72,3 +76,41 @@ def test_step_reporter(power_env, pw_cycle_test, color_scheme, pytest_extra_para
         spawn.expect(pexpect.EOF)
         spawn.close()
         assert spawn.exitstatus == 0
+
+
+@pytest.mark.parametrize(
+    "color_scheme,result_color_scheme,warning_msg",
+    (
+        ("dark", "dark", ""),
+        ("light", "light", ""),
+        ("dark-256color", "dark-256color", ""),
+        ("", "dark", ""),   # the curses_init fixture sets up an 8-color terminal
+        ("non-existing-color-scheme", "dark", "Color scheme 'non-existing-color-scheme' unknown"),
+    )
+)
+def test_step_reporter_color_scheme(caplog, curses_init, color_scheme, result_color_scheme, warning_msg):
+    from labgrid.pytestplugin.reporter import ColoredStepReporter
+
+    class MockTW:
+        _file = None
+    class MockTerminalReporter:
+        def __init__(self, f):
+            self._tw = MockTW()
+            self._tw._file = f
+
+    if color_scheme:
+        os.environ["LG_COLOR_SCHEME"] = color_scheme
+    else:
+        os.environ.pop("LG_COLOR_SCHEME", None)
+
+    caplog.clear()
+
+    with open("/dev/null") as f:
+        csr = ColoredStepReporter(MockTerminalReporter(f))
+    assert csr.color_scheme == ColoredStepReporter.EVENT_COLOR_SCHEMES[result_color_scheme]
+
+    caplog_warnings = [x.message for x in caplog.records if x.levelno == logging.WARNING]
+    if warning_msg:
+        assert warning_msg in caplog_warnings
+    else:
+        assert [] == caplog_warnings
```

Comment:
Internalised behaviour from my C coding days… I like to do it this way more than `var == constant` because it errors out when I accidentally forget one `=` and it becomes an assignment. Your suggestion looks better, so I used it.